README.md

```diff
@@ -32,3 +32,85 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 ```
+
+## Overview
+
+Go Function Coverage tool 'funccover' support generating instrumented source code files
+so that running binary automatically collects the coverage data for functions.
+    
+'funccover' inserts a global function coverage variable that will keep the coverage data for functions 
+to the given source code. Then it inserts necessary instructions to the top of each function 
+(basic assignment instruction to global coverage variable). This way when a function starts executing, 
+global coverage variable will keep the information that this function started execution some time. 
+
+We also have to save that coverage information somewhere. Initially 'funccover' tool writes coverage information
+to a file (RPC will be more useful in the future). Currently 'funccover' tool inserts 2 functions to the given
+source code, one writes coverage data to a file, other calls it periodically. Period must be given as a flag to the tool.
+Tool also inserts a defer call to the main to write coverage data after main function ends. So it is more general. 
+
+## Quickstart
+
+
+
+```bash
+# Get the module from Github and install it into your $GOPATH/bin/
+$ go get github.com/muratekici/go-function-coverage/...
+```
+- If you add your _$GOPATH/bin_ into your _$PATH_ ([instructions](
+https://github.com/golang/go/wiki/GOPATH)) you can run 'funccover' directly by writing 'funccover' to the terminal. 
+
+## How To Use It
+
+```bash
+$ funccover [flags] [args...]
+```
+
+All sources given to funccover must be in the same package
+
+### Flags
+
+'funccover' has 3 flags. Each flag tells 'funccover' how and where it should instrument the source codes. 
+
+#### -period duration
+
+-period flag represents the period of the data collection, if it is not given periodical collection will be disabled. 
+
+```bash
+$ funccover -period=500ms main.go oth.go
+```
+
+#### -dir string
+
+-dir flag sets the destination directory for the instrumented source codes , filenames will be the same.
+If not given it will write instrumented sources to stdout.
+
+```bash
+$ funccover -dir=instrumented_sources source1.go source2.go
+```
+
+#### -o string
+
+-o flag sets the name of the coverage output file name (default "cover.out").
+
+```bash
+$ funccover -period=1s -dst=instrumented.go -o=function_coverage.out source.go
+```
+
+### Example Usage
+
+You have 2 source files named src.go and fun.go, both belongs to the same package. Normally binary runs with 2 arguments. You want to get the function coverage data for the binary to a file named cover.txt and since it is a long running code you want to get the coverage data every 1 minutes.
+
+```bash
+$ funccover -period=1m -o=cover.txt src.go fun.go
```

Comment:
`-dir` is required for what what follows; so perhaps:
`
$ funccover -period=1m -o=cover.txt -dir=instrumented src.go fun.go
`